### PR TITLE
changes to support msvc builds with qt6

### PR DESCRIPTION
**Pull request by jef-n**

### CODE REVIEW

**Comment by m-kuhn** [comment]
I was hoping we could switch to unicode, see https://github.com/qgis/QGIS/pull/56841

File: src/app/CMakeLists.txt [hunk 1]
```diff
@@ -514,6 +514,10 @@ if (ANDROID)
 else()
   add_executable(${QGIS_APP_NAME} MACOSX_BUNDLE WIN32 ${QGIS_APPMAIN_SRCS})
 
+  if(MSVC AND BUILD_WITH_QT6)
+    qt_disable_unicode_defines(${QGIS_APP_NAME})
+  endif(MSVC AND BUILD_WITH_QT6)
```
Comment (m-kuhn) [inline src/app/CMakeLists.txt:519]: Should probably not be added, https://github.com/qgis/QGIS/pull/56980 should fix this

**Comment by m-kuhn** [comment]
Looks good, but I think we should start using the `W`idechar version now that we are touching this code and Qt made this the standard with Qt6

**Comment by MehdiChinoune** [comment]
Could I reopen my PR and merge it first then?

**Pull request closed without merging**